package com.controller;

import com.exception.DivideByZeroException;

public class ExceptionDemo {
	public static void main(String[] args) { //thread: main
		A a=new A(); 
		a.doTask();
		 
		int result;
		try {
			result = a.compute(12,4);
			System.out.println("Result is " + result);
		} catch (DivideByZeroException e) {
			System.out.println("deno cannot be zero");
		}
		
		System.out.println("thank you");
	}
}

class A{ //you are this programmer : service class 
	void doTask() {
		System.out.println("task done");
	}

	public int compute(int i, int j) throws DivideByZeroException{
		  
		if(j == 0) { //validating j and throwing exception
			throw new DivideByZeroException();
		}
		// perform divide
		 return i/j;
	}
}
